app/models/chargeback/consumption_without_rollups.rb

```diff
@@ -29,7 +29,7 @@ def tag_list_with_prefix
     end
 
     def parents_determining_rate
-      [resource.host, resource.ems_cluster, resource.storage, resource.try(:cloud_volumes), parent_ems, resource.tenant,
+      [resource.try(:host), resource.try(:ems_cluster), resource.try(:storage), resource.try(:cloud_volumes), parent_ems, resource.try(:tenant),
```

Comment:
I wonder if we should be asking the `resource` what parents it has instead of trying a bunch of stuff? @lpichler ?

Reply:
probably yes and I am not sure how yet - anyway I am ok with this change. It is also used for [consumption with rollup.](https://github.com/ManageIQ/manageiq/blob/43efe74104398935ff1eaeef9b0c536aec6710fa/app/models/chargeback/consumption_with_rollups.rb#L4)

From rate assignment point of view - only rates assigned to "The Enterprise" and "Selected Provider" can be applied.

Reply:
Perhaps that can be done in a follow-up PR since the timeframe for this is tight